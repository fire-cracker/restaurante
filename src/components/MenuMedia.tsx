import React, { FC, ReactElement } from 'react'
import Col from 'react-bootstrap/Col'
import Media from 'react-bootstrap/Media'

interface IProps {
  imageUrl: string
  name: string
  price: number
  recipe: string
}

const MenuMedia: FC<IProps> = ({ imageUrl, name, price, recipe }): ReactElement => (
  <Col lg={6} className="media-col mb-5">
    <Media>
      <img width={200} height={150} className="mr-3" src={imageUrl} alt={name} />
      <Media.Body className="pt-1">
        <h5 className="mb-3 pb-2 d-flex flex-row justify-content-between">
          {name} <span className="text-darkkhaki">${price}</span>
        </h5>
        <p>{recipe}</p>
      </Media.Body>
    </Media>
  </Col>
)

export default MenuMedia
