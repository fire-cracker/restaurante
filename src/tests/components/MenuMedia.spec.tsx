import React from 'react'
import { render } from '@testing-library/react'

import MenuMedia from '../../components/MenuMedia'

interface IProps {
  imageUrl: string
  name: string
  price: number
  recipe: string
}

describe('MenuMedia', () => {
  const props: IProps = {
    imageUrl: '',
    name: 'Jollof rice',
    price: 300,
    recipe: 'green pepper'
  }

  let wrapper: any
  beforeEach(() => {
    wrapper = render(<MenuMedia {...props} />)
  })

  afterAll(() => {
    jest.clearAllMocks()
  })

  test('renders MenuMedia component', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('renders menu name', () => {
    const menuName = wrapper.getByText('Jollof rice')
    expect(menuName).toBeInTheDocument()
  })

  test('renders menu price', () => {
    const menuPrice = wrapper.getByText('$300')
    expect(menuPrice).toBeInTheDocument()
  })

  test('renders menu recipe', () => {
    const menuRecipe = wrapper.getByText('green pepper')
    expect(menuRecipe).toBeInTheDocument()
  })
})
