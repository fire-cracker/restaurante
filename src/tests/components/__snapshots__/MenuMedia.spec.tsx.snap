// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuMedia renders MenuMedia component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="media-col mb-5 col-lg-6"
      >
        <div
          class="media"
        >
          <img
            alt="Jollof rice"
            class="mr-3"
            height="150"
            src=""
            width="200"
          />
          <div
            class="pt-1 media-body"
          >
            <h5
              class="mb-3 pb-2 d-flex flex-row justify-content-between"
            >
              Jollof rice
               
              <span
                class="text-darkkhaki"
              >
                $
                300
              </span>
            </h5>
            <p>
              green pepper
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="media-col mb-5 col-lg-6"
    >
      <div
        class="media"
      >
        <img
          alt="Jollof rice"
          class="mr-3"
          height="150"
          src=""
          width="200"
        />
        <div
          class="pt-1 media-body"
        >
          <h5
            class="mb-3 pb-2 d-flex flex-row justify-content-between"
          >
            Jollof rice
             
            <span
              class="text-darkkhaki"
            >
              $
              300
            </span>
          </h5>
          <p>
            green pepper
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
